src/store/wrapper.rs

```diff
@@ -45,6 +47,7 @@ where
     }
 }
 
+// TODO could this be a blanket implementation for every S that returns dyn error
```

Comment:
```suggestion
```

The error could be re-worked if we change the Store trait. I don't think an extension trait would be needed though.